router/utils/types.go

```diff
@@ -0,0 +1,39 @@
+package utils
+
+import (
+	"reflect"
+	"strings"
+)
+
+// GetType returns reflect.Type of the generic parameter it receives.
+func GetType[T any]() reflect.Type { return reflect.TypeOf(new(T)).Elem() }
+
+// Nil represents an empty type.
+// You can use it with the HandlerFunc generic parameters to declare no Request with no request body, no path or query
+// params, or responses with no response body.
+type Nil *uintptr
+
+// NilType represent the type of Nil.
+var NilType = GetType[Nil]()
+
+const ignoreFieldTagValue = "-"
+
+func StructKeys(structType reflect.Type, tag string) map[string]reflect.StructField {
```

Comment:
This function is difficult to understand (for instance, the recursion for embedded objects is hidden and not explained.)
Can you refactor a little, or add documentation?

Reply:
Added documentation for the behavior of this function.
Additionally, you can check `TestStructKeys`.